spython/utils/terminal.py

```diff
@@ -103,9 +103,14 @@ def stream_command(cmd, no_newline_regexp="Progess", sudo=False):
        cmd: the command to send, should be a list for subprocess
        no_newline_regexp: the regular expression to determine skipping a
                           newline. Defaults to finding Progress
+       sudo_options: string or list of strings that will be passed as options to sudo
 
     '''
-    if sudo:
+    if sudo and sudo_options is not None:
+        if isinstance(sudo_options, str):
+            sudo_options = sudo_options.split()
```

Comment:
For this, I think we should use shlex to parse the string. See shlex.split https://docs.python.org/3/library/shlex.html

Reply:
Done